core/server/api_container/server/service_network/default_service_network.go

```diff
@@ -1791,3 +1878,102 @@ func renderTemplateToFile(templateAsAString string, templateData interface{}, de
 	}
 	return nil
 }
+
+func waitUntilAllTCPAndUDPPortsAreOpen(
+	ipAddr net.IP,
+	ports map[string]*port_spec.PortSpec,
+) error {
+
+	var portCheckErrorGroup errgroup.Group
+
+	for _, portSpec := range ports {
+		wrappedWaitFunc := func() error {
+			return waitUntilPortIsOpenWithTimeout(ipAddr, portSpec)
+		}
+		portCheckErrorGroup.Go(wrappedWaitFunc)
+	}
+	//This error group pattern allow us to reject early if at least one of the ports check fails
+	//in this opportunity we want to reject early because the main user pain that we want to handle
+	//is a wrong service configuration
+	if err := portCheckErrorGroup.Wait(); err != nil {
+		return stacktrace.Propagate(err, "An error occurred while waiting for all TCP and UDP ports to be open")
+	}
+
+	return nil
+}
+
+func waitUntilPortIsOpenWithTimeout(
+	ipAddr net.IP,
+	portSpec *port_spec.PortSpec,
+) error {
+	// reject early if it's disable
+	if !portSpec.GetWait().GetEnable() {
+		return nil
+	}
+	timeout := portSpec.GetWait().GetTimeout()
+
+	var (
+		ticker                  = time.NewTicker(timeout)
```

Comment:
Don't forget to defer a Stop for the ticker. From the Ticker doc:
```
Stop the ticker to release associated resources.
```

Reply:
good catch!! thanks